build.gradle

```diff
@@ -80,6 +81,30 @@ targetCompatibility = 1.11
 
 test {
     useJUnitPlatform()
+    finalizedBy(['jacocoTestReport', 'jacocoTestCoverageVerification'])
+}
+
+jacoco {
+    toolVersion = "0.8.5"
+    reportsDir = file("$buildDir/reports/jacoco")
+}
+
+jacocoTestReport {
+    reports {
+        xml.enabled false
+        csv.enabled false
+        html.destination file("${buildDir}/reports/jacoco")
+    }
+}
+
+jacocoTestCoverageVerification {
+    violationRules {
+        rule {
+            limit {
+                minimum = 0.72
```

Comment:
❓These questions shouldn't block this PR. I'm just curious. How does this work for submodules? Is all of the java code analyzed regardless of which module it's in and that all is compiled into one report? I know the solr submodule doesn't have any tests in it now, but if it did, would it be analyzed?

Reply:
I'm not exactly sure, but I'd assume it would work by just overriding these rules in the submodules gradle file by redefining this task for that sub module.

Reply:
So each module would have its own jacoco task, something like `:module1:jacocoTestCoverageVerification` and `:module2:jacocoTestCoverageVerification`. It's possible to tell the top level gradle file to run certain tasks on all sub modules